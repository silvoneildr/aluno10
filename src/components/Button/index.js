import React from 'react';
import propTypes from 'prop-types';

import StyleButton from './styles';

const Button = React.memo(({ children, handleClick, ...rest }) => (
  <StyleButton onClick={handleClick} {...rest}>
    {children}
  </StyleButton>
));

Button.propTypes = {
  children: propTypes.string.isRequired,
  handleClick: propTypes.func.isRequired
};

export default Button;
