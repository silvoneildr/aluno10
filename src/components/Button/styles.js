import styled, { css } from 'styled-components';

const StyleButton = styled.button`
  width: ${props => props.width || '100px'};
  height: ${props => props.height || '50px'};
  color: white;
  border-radius: 4px;

  ${props =>
    props.color &&
    css`
      background: ${props.theme[props.color] || props.theme.primary};
    `}
`;

export default StyleButton;
