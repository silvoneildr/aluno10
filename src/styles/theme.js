export default {
  strong: '#052c4b',
  primary: '#4548fd',
  secondary: '#ffffff',
  light: '#dcdee091',
  success: '#37da6b',
  warning: '#f0d74d',
  error: '#f8562d'
};
