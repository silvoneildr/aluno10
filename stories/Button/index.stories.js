import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import { Wrapper } from '../index.stories';
import Button from '../../src/components/Button';

storiesOf('Components|Button', module).add('primary', () => (
  <Wrapper>
    <Button handleClick={action('clicked')} color="primary">
      Hello Button
    </Button>
  </Wrapper>
));

storiesOf('Components|Button', module).add('success', () => (
  <Wrapper>
    <Button handleClick={action('clicked')} color="success">
      Hello Button
    </Button>
  </Wrapper>
));

storiesOf('Components|Button', module).add('error', () => (
  <Wrapper>
    <Button handleClick={action('clicked')} color="error">
      Hello Button
    </Button>
  </Wrapper>
));
