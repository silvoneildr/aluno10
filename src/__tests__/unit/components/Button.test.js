import React from 'react';
import Button from 'Components/Button';
import theme from 'Root/styles/theme';
import { shallow } from 'enzyme';
import { render, cleanup } from 'react-testing-library';

let wrapper;

afterEach(() => {
  if (wrapper) {
    wrapper.unmount();
  }
  cleanup();
});

describe('Button', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Button onClick={() => {}} theme={theme}>
        Hello World
      </Button>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should handle click correctly', () => {
    const click = jest.fn();
    wrapper = shallow(
      <Button onClick={click} theme={theme}>
        Hello World
      </Button>
    );

    wrapper.props().onClick();
    expect(click).toHaveBeenCalled();
  });
});
