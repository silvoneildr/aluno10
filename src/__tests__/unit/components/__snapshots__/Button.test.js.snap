// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render button correctly 1`] = `
.c0 {
  width: 100px;
  height: 50px;
  color: white;
  border-radius: 4px;
}

<Component
  handleClick={[Function]}
  theme={
    Object {
      "error": "#f8562d",
      "light": "#dcdee091",
      "primary": "#4548fd",
      "secondary": "#ffffff",
      "strong": "#052c4b",
      "success": "#37da6b",
      "warning": "#f0d74d",
    }
  }
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  width: ",
            [Function],
            ";
  height: ",
            [Function],
            ";
  color: white;
  border-radius: 4px;

  ",
            [Function],
            "
",
          ],
        },
        "displayName": "styled.button",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "button",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    onClick={[Function]}
    theme={
      Object {
        "error": "#f8562d",
        "light": "#dcdee091",
        "primary": "#4548fd",
        "secondary": "#ffffff",
        "strong": "#052c4b",
        "success": "#37da6b",
        "warning": "#f0d74d",
      }
    }
  >
    <button
      className="c0"
      onClick={[Function]}
    >
      Hello World
    </button>
  </StyledComponent>
</Component>
`;

exports[`Button should render primary button correctly 1`] = `
.c0 {
  width: 100px;
  height: 50px;
  color: white;
  border-radius: 4px;
}

<Component
  handleClick={[MockFunction]}
  theme={
    Object {
      "error": "#f8562d",
      "light": "#dcdee091",
      "primary": "#4548fd",
      "secondary": "#ffffff",
      "strong": "#052c4b",
      "success": "#37da6b",
      "warning": "#f0d74d",
    }
  }
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  width: ",
            [Function],
            ";
  height: ",
            [Function],
            ";
  color: white;
  border-radius: 4px;

  ",
            [Function],
            "
",
          ],
        },
        "displayName": "styled.button",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "button",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    onClick={[MockFunction]}
    theme={
      Object {
        "error": "#f8562d",
        "light": "#dcdee091",
        "primary": "#4548fd",
        "secondary": "#ffffff",
        "strong": "#052c4b",
        "success": "#37da6b",
        "warning": "#f0d74d",
      }
    }
  >
    <button
      className="c0"
      onClick={[MockFunction]}
    >
      Hello World
    </button>
  </StyledComponent>
</Component>
`;
